ygen/gogen.go

```diff
@@ -1151,6 +1284,76 @@ func generateValidator(buf *bytes.Buffer, structDef generatedGoStruct) error {
 	return nil
 }
 
+// generateGetOrCreateStruct generates a getter method for the YANG container
+// (Go struct ptr) fields of structDef, and appends it to the supplied buffer.
+// Assuming that structDef represents the following struct:
+//
+//  type MyStruct struct {
+// 		Container *MyStruct_Container
+//  }
+//
+// the getter function generated for the struct will be:
+//
+//  func (s *MyStruct) GetOrCreateContainer() *MyStruct_Container {
+//    if s.Container != nil {
+//      return s.Container
+//    }
+//    s.Container = &MyStruct_Container{}
+//    return s.Container
+//  }
+func generateGetOrCreateStruct(buf *bytes.Buffer, structDef generatedGoStruct) error {
+	type structFieldWithName struct {
+		StructName string
+		Field      *goStructField
+	}
+	for _, f := range structDef.Fields {
+		if f.IsYANGContainer {
+			tmpStruct := structFieldWithName{
+				StructName: structDef.StructName,
+				Field:      f,
+			}
+			if err := goTemplates["getOrCreateStruct"].Execute(buf, tmpStruct); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// generateGetOrCreateList generates a getter function similar to that created
+// by the generateGetOrCreateStruct function for maps within the generated Go
+// code (which represent YANG lists). It handles both simple and composite key
+// lists.
+//
+// If the list described has a single key, the argument to the function is the
+// non-pointer key value. If the list has a complex type, it is a instance of
+// the generated key type for the list.
+//
+// The generated function returns the existing value if the key exists in the
+// list, or creates a new value using the NewXXX method if it does not exist.
+// The generated function is written to the supplied buffer, using the method
+// argument to determine the list's characteristics in the template.
+func generateGetOrCreateList(buf *bytes.Buffer, method *generatedGoListMethod) error {
+	if err := goTemplates["getOrCreateList"].Execute(buf, method); err != nil {
```

Comment:
Can this and the following function just return directly?

return gotemplates[...]...

Reply:
Yes, not sure why I wrote this like this -- fixed.